### ISSUE TITLE
[v5-rc] Query.In with search by BsonValue does not work

### ISSUE THREAD

**Issue by Tirraon** [mbdavid/LiteDB#1419]
With latest NuGet package and latest master I have noticed, that DeleteMany does not work anymore, when I try to filter ids by guid array. After some research, I have found, that problem is in Query.In. 
It seems like it doesn't return any result, when I try to filter by Guid.

Minimal repro
```csharp
public class UserGuid
{
    public Guid Id { get; set; }
    public string Name { get; set; }
}

[Fact]
public void Query_In_By_Guid()
{
    using (var db = new LiteDatabase(new MemoryStream()))
    {
        var users = db.GetCollection<UserGuid>("users_guid");

        var u1 = new UserGuid { Id = Guid.NewGuid() , Name = "John" };
        var u2 = new UserGuid { Id = Guid.NewGuid(), Name = "Zarlos" };
        var u3 = new UserGuid { Id = Guid.NewGuid(), Name = "Ana" };

        users.Insert(new UserGuid[] { u1, u2, u3 });

        var ids = new Guid[] { u1.Id, u2.Id, u3.Id }.Select(i => new BsonValue(i));

        users.Find(Query.In(nameof(UserGuid.Id), ids)).Should().HaveCount(3); // actual == 0
    }
}
```

Workaround - use expressions
```csharp
var ids = new Guid[] { u1.Id, u2.Id, u3.Id };
users.Find(u => ids.Contains(u.Id)).Should().HaveCount(3); // actual == 3
```

**Comment by Tirraon** [comment]
Just noticed, that Query.In doesn't work for any type.
With integer Id results are same.

Probably one previous issue was about that problem https://github.com/mbdavid/LiteDB/issues/1401

**Comment by lbnascimento** [comment]
@Tirraon There was a bug in the execution of IndexIn, it was fixed on master.

Also, you should be using `"_id"` instead of `nameof(UserGuid.Id)` for the field name. The id field is always becomes `"_id"` in the serialized document.

**Comment by JensSchadron** [comment]
Just FYI, this was fixed in #1420 :)

**Comment by Tirraon** [comment]
@lbnascimento 
@JensSchadron 
Not sure about "_id", because nameof() solution have worked for me earlier. Or I am just missing something and I have filtered not by id-field...

By the way, Query.In("_id", ids) works now, so I will close this issue.
Thanks!
